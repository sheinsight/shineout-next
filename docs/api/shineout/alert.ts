const api = JSON.parse('[{"title":"Alert","properties":[{"name":"className","tag":{"cn":"自定义类名","en":"Custom class name","default":"","version":""},"required":false,"type":"string "},{"name":"style","tag":{"cn":"自定义样式","en":"Custom style","default":"","version":""},"required":false,"type":"CSSProperties "}],"cn":"","en":""}]');
export default api;
