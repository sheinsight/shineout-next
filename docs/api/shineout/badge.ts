const api = JSON.parse('[{"title":"Badge","properties":[{"name":"className","tag":{"cn":"自定义类名","en":"Custom class name","default":"","version":""},"required":false,"type":"string "},{"name":"style","tag":{"cn":"自定义样式","en":"Custom style","default":"","version":""},"required":false,"type":"CSSProperties "},{"name":"children","tag":{"cn":"内容","en":"Content","default":"","version":""},"required":false,"type":"ReactNode"},{"name":"color","tag":{"cn":"自定义徽标颜色","en":"Custom badge color","default":"","version":""},"required":false,"type":"string "},{"name":"count","tag":{"cn":"展示的数字，大于 overflowCount 时显示 ${overflowCount}+，为 0 时隐藏","en":"Number to show, show ${overflowCount}+ when it is greater than overflowCount, hide when it is 0","default":"","version":""},"required":false,"type":"ReactNode"},{"name":"dot","tag":{"cn":"小点模式，开启后不展示数字","en":"Dot mode","default":"false","version":""},"required":false,"type":"boolean "},{"name":"offset","tag":{"cn":"偏移量","en":"Offset of the badge","default":"","version":""},"required":false,"type":"[number, number] "},{"name":"overflowCount","tag":{"cn":"封顶数值","en":"Offset of the badge","default":"","version":""},"required":false,"type":"number "},{"name":"showZero","tag":{"cn":"当数值为 0 时，是否展示 Badge","en":"Whether to show Badge when the number is 0","default":"","version":""},"required":false,"type":"boolean "},{"name":"status","tag":{"cn":"徽标状态","en":"Whether to show Badge when the number is 0","default":"","version":""},"required":false,"type":"\\\"success\\\" | \\\"processing\\\" | \\\"default\\\" | \\\"error\\\" | \\\"warning\\\" "},{"name":"text","tag":{"cn":"状态点文本，仅在配置了 status 属性后生效","en":"Status point text, only valid after configuring the status property","default":"","version":""},"required":false,"type":"ReactNode"},{"name":"size","tag":{"cn":"尺寸，仅在非 dot 模式下生效，支持 small 和 default 两种尺寸","en":"Size, only valid in non-dot mode, supports two sizes: small and default","default":"","version":""},"required":false,"type":"\\\"default\\\" | \\\"small\\\" "}],"cn":"","en":"","sort":"0"}]');
export default api;
