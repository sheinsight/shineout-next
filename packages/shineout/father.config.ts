import { defineConfig } from 'father';
export default defineConfig({
  extends: '../../.fatherrc.ts',
});
