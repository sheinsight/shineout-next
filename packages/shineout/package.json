{
  "name": "shineout",
  "version": "3.0.0",
  "description": "A components library for React",
  "keywords": [],
  "license": "ISC",
  "author": "",
  "sideEffects": false,
  "main": "./dist/cjs/index.js",
  "module": "./dist/esm/index.js",
  "typings": "./dist/cjs/index.d.ts",
  "scripts": {
    "build": "father build",
    "build:watch": "father dev",
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "dependencies": {
    "@sheinx/base": "workspace:*",
    "@sheinx/hooks": "workspace:*",
    "@sheinx/shineout-style": "workspace:*"
  },
  "peerDependencies": {
    "classnames": ">=2",
    "immer": ">=10",
    "react": ">=16.8"
  },
  "devDependencies": {
    "jszip": "^3.10.1"
  }
}
