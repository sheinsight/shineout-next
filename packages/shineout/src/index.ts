export { default as Alert } from './alert';
export { default as Button } from './button';
export { default as Checkbox } from './checkbox';
export { default as DatePicker } from './date-picker';
export { default as Dropdown } from './dropdown';
export { default as EditableArea } from './editable-area';
export { default as Form } from './form';
export { default as Icon } from './icon';
export { default as Image } from './image';
export { default as Input } from './input';
export { default as Popover } from './popover';
export { default as Radio } from './radio';
export { default as Rule } from './rule';
export { default as Spin } from './spin';
export { default as Switch } from './switch';
export { default as Tag } from './tag';
export { default as Tabs } from './tabs';
export { default as Textarea } from './textarea';
export { default as Tooltip } from './tooltip';
export { setConfig } from '@sheinx/base';

export * as TYPE from './type';
