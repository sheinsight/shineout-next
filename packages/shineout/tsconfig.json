{
  "extends": "../../tsconfig.json",
  "compilerOptions": {
    "declaration": true,
    "baseUrl": "./",
    "paths": {
      "shineout": ["./src/index"],
      "@sheinx/base": ["../base/src/index"],
      "@sheinx/hooks": ["../hooks/src/index"]
    }
  },
  "include": ["src"]
}
