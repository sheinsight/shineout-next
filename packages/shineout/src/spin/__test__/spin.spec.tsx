import { useState } from 'react';
import { render, cleanup, fireEvent } from '@testing-library/react';
import Spin from '..';
import { Button } from 'shineout';
import {
  snapshotTest,
  styleTest,
  classTest,
  textContentTest,
  displayTest,
  baseTest,
  childrenTest,
} from '../../tests/utils';
import mountTest from '../../tests/mountTest';
import { classLengthTest } from '../../tests/structureTest';
import SpinBase from '../__example__/01-base-1';
import SpinBaseAll from '../__example__/01-base-2';
import SpinMode from '../__example__/02-mode';
import SpinTip from '../__example__/04-tip';

const SO_PREFIX = 'spin';
const spinDefaultClassName = `${SO_PREFIX}-default-0-2-20`;
const spinClassName = `.${SO_PREFIX}-spin-0-2-36`;
const spinRingClassName = `.${SO_PREFIX}-ring-0-2-31`;
const spinItemClassName = `.${SO_PREFIX}-item-0-2-25`;
const spinChasingDotsClassName = `${SO_PREFIX}-chasingDots-0-2-21`;
const spinCubeGridClassName = `${SO_PREFIX}-cubeGrid-0-2-22`;
const spinDoubleBounceClassName = `${SO_PREFIX}-doubleBounce-0-2-23`;
const spinFadingCircleClassName = `${SO_PREFIX}-fadingCircle-0-2-26`;
const spinFadeClassName = `${SO_PREFIX}-fade-0-2-35`;
const spinFourDotsClassName = `${SO_PREFIX}-fourDots-0-2-28`;
const spinPlaneClassName = `.${SO_PREFIX}-plane-0-2-29`;
const spinPulseClassName = `.${SO_PREFIX}-pulse-0-2-30`;
const spinThreeBounceClassName = `${SO_PREFIX}-threeBounce-0-2-32`;
const spinWaveClassName = `${SO_PREFIX}-wave-0-2-33`;
const spinChasingRingClassName = `${SO_PREFIX}-chasingRing-0-2-34`;
const spinContentClassName = `.${SO_PREFIX}-content-0-2-37`;
const spinVerticalClassName = `${SO_PREFIX}-vertical-0-2-41`;
const spinTipClassName = `.${SO_PREFIX}-tip-0-2-40`;
const spinHorizontalClassName = `${SO_PREFIX}-horizontal-0-2-42`;
const spinContainerClassName = `.${SO_PREFIX}-container-0-2-38`;
const spinLoadingClassName = `.${SO_PREFIX}-loading-0-2-60`;
const spinLoadingOtherClassName = `.${SO_PREFIX}-loading-0-2-39`;
const spinRingOtherClassName = `.${SO_PREFIX}-ring-0-2-52`;

type SpinNameType =
  | 'default'
  | 'chasing-dots'
  | 'cube-grid'
  | 'double-bounce'
  | 'fading-circle'
  | 'four-dots'
  | 'plane'
  | 'pulse'
  | 'ring'
  | 'scale-circle'
  | 'three-bounce'
  | 'wave'
  | 'chasing-ring';

const spinObj: { [key: string]: (string | number)[] } = {
  default: [spinDefaultClassName, 12],
  'chasing-dots': [spinChasingDotsClassName, 2],
  'cube-grid': [spinCubeGridClassName, 9],
  'double-bounce': [spinDoubleBounceClassName, 2],
  'fading-circle': [spinFadingCircleClassName, 12],
  'four-dots': [spinFourDotsClassName, 4],
  'scale-circle': [spinFadingCircleClassName, 12],
  wave: [spinWaveClassName, 5],
  'chasing-ring': [spinChasingRingClassName, 4],
};
const spinObjWithoutItem: { [key: string]: string } = {
  plane: spinPlaneClassName,
  pulse: spinPulseClassName,
};
const size = 20;
const sizeFS = '30px';

// converted to number when set size is string
const toNumber = (size: number | string) =>
  typeof size === 'number' ? size : Number(size.slice(0, -2));
const styleRenderXY = (size: number | string) =>
  `width: ${toNumber(size)}px; height: ${toNumber(size)}px;`;
const styleRender = (size: number | string) =>
  `${styleRenderXY(size)} border-width: ${toNumber(size) / 10}px; font-size: ${
    toNumber(size) / 10
  }px;`;
const styleAuto = (size: number | string) => `width: ${toNumber(size) * 2}px; height: auto;`;
const style = styleRenderXY(40);

afterEach(cleanup);
mountTest(Spin);
describe('Spin[Base]', () => {
  displayTest(Spin, 'ShineoutSpin');
  baseTest(Spin, spinClassName, { backgroundColor: 'red' }, style + ' background-color: red;');
  childrenTest(Spin, spinContainerClassName);
  snapshotTest(<SpinBase />);
  snapshotTest(<SpinBaseAll />);
  test('should render when set color', () => {
    const color = 'red';
    const { container } = render(<Spin color={color} />);
    const items = container.querySelectorAll(spinItemClassName);
    items.forEach((item) => {
      expect(
        item.querySelector('div')?.getAttribute('style')?.includes(`background-color: ${color};`),
      ).toBeTruthy();
    });
  });
});
describe('Spin[Name/Size]', () => {
  Object.keys(spinObj).forEach((name) => {
    test(`should render when set name is ${name}`, () => {
      const { container, rerender } = render(<Spin name={name as SpinNameType} />);
      const spin = container.querySelector(spinClassName)!;
      classTest(spin, spinObj[name][0] as string);
      styleTest(spin, style);
      classLengthTest(spin, spinItemClassName, spinObj[name][1] as number);
      rerender(<Spin name={name as SpinNameType} size={size} />);
      styleTest(spin, styleRenderXY(size));
      rerender(<Spin name={name as SpinNameType} size={sizeFS} />);
      styleTest(spin, styleRenderXY(sizeFS));
      if (name !== 'fading-circle') return;
      container.querySelectorAll(spinItemClassName).forEach((item) => {
        classTest(item, spinFadeClassName);
      });
    });
  });
  Object.keys(spinObjWithoutItem).forEach((name) => {
    test(`should render when set name is ${name}`, () => {
      const { container, rerender } = render(<Spin name={name as SpinNameType} />);
      const spin = container.querySelector(spinObjWithoutItem[name])!;
      expect(spin).toBeTruthy();
      styleTest(spin, style);
      rerender(<Spin name={name as SpinNameType} size={size} />);
      styleTest(spin, styleRenderXY(size));
      rerender(<Spin name={name as SpinNameType} size={sizeFS} />);
      styleTest(spin, styleRenderXY(sizeFS));
    });
  });
  test('should render when set name is ring', () => {
    const { container, rerender } = render(<Spin name='ring' />);
    const spin = container.querySelector(spinRingClassName)!;
    expect(spin).toBeTruthy();
    styleTest(spin, styleRender(40));
    rerender(<Spin name='ring' size={size} />);
    styleTest(spin, styleRender(size));
    rerender(<Spin name='ring' size={sizeFS} />);
    styleTest(spin, styleRender(sizeFS));
  });
  test('should render when set name is three-bounce', () => {
    const { container, rerender } = render(<Spin name='three-bounce' />);
    const spin = container.querySelector(spinClassName)!;
    classTest(spin, spinThreeBounceClassName);
    styleTest(spin, 'width: 80px; height: auto;');
    classLengthTest(spin, spinItemClassName, 3);
    rerender(<Spin name='three-bounce' size={size} />);
    styleTest(spin, styleAuto(size));
    rerender(<Spin name='three-bounce' size={sizeFS} />);
    styleTest(spin, styleAuto(sizeFS));
  });
});
describe('Spin[Tip]', () => {
  snapshotTest(<SpinTip />);
  const tipTest = (container: Element, text: string) => {
    const spin = container.querySelector(spinContentClassName)!;
    classTest(spin, spinVerticalClassName);
    expect(spin.querySelector(spinTipClassName)).toBeTruthy();
    textContentTest(spin.querySelector(spinTipClassName)!, text);
  };
  test('should render when set tip is string', () => {
    const text = 'loading';
    const { container } = render(<Spin name={'ring' as SpinNameType} tip={text} />);
    tipTest(container, text);
  });
  test('should rende when set tip is function', () => {
    const { container } = render(<SpinTip />);
    tipTest(container, 'This may take a while...');
  });
});
describe('Spin[Mode]', () => {
  snapshotTest(<SpinMode />);
  test('should render when set mode', () => {
    const { container } = render(<SpinMode />);
    const spins = container.querySelectorAll(spinContentClassName);
    classTest(spins[0], spinVerticalClassName);
    classTest(spins[1], spinHorizontalClassName);
  });
});
describe('Spin[inexistence]', () => {
  test('should render when set name is inexistence', () => {
    const inexistenceName = 'hello';
    const { container } = render(<Spin name={inexistenceName as SpinNameType} />);
    classLengthTest(container, spinClassName, 0);
  });
  test('should render when set name is undefined', () => {
    const { container } = render(<Spin name={undefined} />);
    classLengthTest(container, spinClassName, 1);
    classTest(container.querySelector(spinClassName)!, spinDefaultClassName);
  });
});
describe('Spin[Loading]', () => {
  test('should render when set loading', () => {
    const Demo = () => {
      const [load, setLoad] = useState<boolean>(false);
      return (
        <div>
          <Button
            onClick={() => {
              setLoad(!load);
            }}
          >
            Click
          </Button>
          <Spin name='ring' loading={load}>
            <div>Demo</div>
          </Spin>
        </div>
      );
    };
    const { container } = render(<Demo />);
    const buttonCon = container.querySelector('button') as HTMLButtonElement;
    expect(
      container.querySelector(spinLoadingClassName) ||
        container.querySelector(spinLoadingOtherClassName),
    ).toBeFalsy();
    fireEvent.click(buttonCon);
    expect(
      container.querySelector(spinLoadingClassName) ||
        container.querySelector(spinLoadingOtherClassName),
    ).toBeTruthy();
  });
  test('should render when set loading without children', () => {
    const Demo = () => {
      const [load, setLoad] = useState<boolean>(false);
      return (
        <div>
          <Button
            onClick={() => {
              setLoad(!load);
            }}
          >
            Click
          </Button>
          <Spin name='ring' loading={load} />
        </div>
      );
    };
    const { container } = render(<Demo />);
    const buttonCon = container.querySelector('button') as HTMLButtonElement;
    expect(
      container.querySelector(spinRingOtherClassName) || container.querySelector(spinRingClassName),
    ).toBeTruthy();
    fireEvent.click(buttonCon);
    expect(
      container.querySelector(spinRingOtherClassName) || container.querySelector(spinRingClassName),
    ).toBeTruthy();
  });
});
