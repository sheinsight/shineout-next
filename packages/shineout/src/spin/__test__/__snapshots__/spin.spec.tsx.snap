// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spin[Base] should render correctly  1`] = `
<div
  style="width: 20px;"
>
  <div
    class="spin-ring-0-2-31"
    style="width: 16px; height: 16px; border-width: 1.6px; font-size: 1.6px;"
  />
</div>
`;

exports[`Spin[Base] should render correctly  2`] = `
<div
  style="display: flex; flex-wrap: wrap; gap: 65px;"
>
  <div
    class="spin-content-0-2-37 spin-vertical-0-2-41"
  >
    <div
      class="spin-default-0-2-20 spin-spin-0-2-36"
      style="width: 16px; height: 16px;"
    >
      <div
        class="spin-item-0-2-25"
      >
        <div
          style="width: 2px; border-radius: 2px;"
        />
      </div>
      <div
        class="spin-item-0-2-25"
      >
        <div
          style="width: 2px; border-radius: 2px;"
        />
      </div>
      <div
        class="spin-item-0-2-25"
      >
        <div
          style="width: 2px; border-radius: 2px;"
        />
      </div>
      <div
        class="spin-item-0-2-25"
      >
        <div
          style="width: 2px; border-radius: 2px;"
        />
      </div>
      <div
        class="spin-item-0-2-25"
      >
        <div
          style="width: 2px; border-radius: 2px;"
        />
      </div>
      <div
        class="spin-item-0-2-25"
      >
        <div
          style="width: 2px; border-radius: 2px;"
        />
      </div>
      <div
        class="spin-item-0-2-25"
      >
        <div
          style="width: 2px; border-radius: 2px;"
        />
      </div>
      <div
        class="spin-item-0-2-25"
      >
        <div
          style="width: 2px; border-radius: 2px;"
        />
      </div>
      <div
        class="spin-item-0-2-25"
      >
        <div
          style="width: 2px; border-radius: 2px;"
        />
      </div>
      <div
        class="spin-item-0-2-25"
      >
        <div
          style="width: 2px; border-radius: 2px;"
        />
      </div>
      <div
        class="spin-item-0-2-25"
      >
        <div
          style="width: 2px; border-radius: 2px;"
        />
      </div>
      <div
        class="spin-item-0-2-25"
      >
        <div
          style="width: 2px; border-radius: 2px;"
        />
      </div>
    </div>
    <div
      class="spin-tip-0-2-40"
    >
      <span
        style="font-size: 12px; font-weight: 300;"
      >
        default
      </span>
    </div>
  </div>
  <div
    class="spin-content-0-2-37 spin-vertical-0-2-41"
  >
    <div
      class="spin-chasingDots-0-2-21 spin-spin-0-2-36"
      style="width: 16px; height: 16px;"
    >
      <div
        class="spin-item-0-2-25"
      >
        <svg
          viewBox="0 0 100 100"
        >
          <circle
            cx="50"
            cy="50"
            r="50"
          />
        </svg>
      </div>
      <div
        class="spin-item-0-2-25"
      >
        <svg
          viewBox="0 0 100 100"
        >
          <circle
            cx="50"
            cy="50"
            r="50"
          />
        </svg>
      </div>
    </div>
    <div
      class="spin-tip-0-2-40"
    >
      <span
        style="font-size: 12px; font-weight: 300;"
      >
        chasing-dots
      </span>
    </div>
  </div>
  <div
    class="spin-content-0-2-37 spin-vertical-0-2-41"
  >
    <div
      class="spin-cubeGrid-0-2-22 spin-spin-0-2-36"
      style="width: 16px; height: 16px;"
    >
      <div
        class="spin-item-0-2-25"
      />
      <div
        class="spin-item-0-2-25"
      />
      <div
        class="spin-item-0-2-25"
      />
      <div
        class="spin-item-0-2-25"
      />
      <div
        class="spin-item-0-2-25"
      />
      <div
        class="spin-item-0-2-25"
      />
      <div
        class="spin-item-0-2-25"
      />
      <div
        class="spin-item-0-2-25"
      />
      <div
        class="spin-item-0-2-25"
      />
    </div>
    <div
      class="spin-tip-0-2-40"
    >
      <span
        style="font-size: 12px; font-weight: 300;"
      >
        cube-grid
      </span>
    </div>
  </div>
  <div
    class="spin-content-0-2-37 spin-vertical-0-2-41"
  >
    <div
      class="spin-doubleBounce-0-2-23 spin-spin-0-2-36"
      style="width: 16px; height: 16px;"
    >
      <div
        class="spin-item-0-2-25"
      />
      <div
        class="spin-item-0-2-25"
      />
    </div>
    <div
      class="spin-tip-0-2-40"
    >
      <span
        style="font-size: 12px; font-weight: 300;"
      >
        double-bounce
      </span>
    </div>
  </div>
  <div
    class="spin-content-0-2-37 spin-vertical-0-2-41"
  >
    <div
      class="spin-fadingCircle-0-2-26 spin-spin-0-2-36"
      style="width: 16px; height: 16px;"
    >
      <div
        class="spin-item-0-2-25 spin-fade-0-2-35"
      >
        <svg
          height="2.286px"
          viewBox="0 0 100 100"
          width="2.286px"
        >
          <circle
            cx="50"
            cy="50"
            r="50"
          />
        </svg>
      </div>
      <div
        class="spin-item-0-2-25 spin-fade-0-2-35"
      >
        <svg
          height="2.286px"
          viewBox="0 0 100 100"
          width="2.286px"
        >
          <circle
            cx="50"
            cy="50"
            r="50"
          />
        </svg>
      </div>
      <div
        class="spin-item-0-2-25 spin-fade-0-2-35"
      >
        <svg
          height="2.286px"
          viewBox="0 0 100 100"
          width="2.286px"
        >
          <circle
            cx="50"
            cy="50"
            r="50"
          />
        </svg>
      </div>
      <div
        class="spin-item-0-2-25 spin-fade-0-2-35"
      >
        <svg
          height="2.286px"
          viewBox="0 0 100 100"
          width="2.286px"
        >
          <circle
            cx="50"
            cy="50"
            r="50"
          />
        </svg>
      </div>
      <div
        class="spin-item-0-2-25 spin-fade-0-2-35"
      >
        <svg
          height="2.286px"
          viewBox="0 0 100 100"
          width="2.286px"
        >
          <circle
            cx="50"
            cy="50"
            r="50"
          />
        </svg>
      </div>
      <div
        class="spin-item-0-2-25 spin-fade-0-2-35"
      >
        <svg
          height="2.286px"
          viewBox="0 0 100 100"
          width="2.286px"
        >
          <circle
            cx="50"
            cy="50"
            r="50"
          />
        </svg>
      </div>
      <div
        class="spin-item-0-2-25 spin-fade-0-2-35"
      >
        <svg
          height="2.286px"
          viewBox="0 0 100 100"
          width="2.286px"
        >
          <circle
            cx="50"
            cy="50"
            r="50"
          />
        </svg>
      </div>
      <div
        class="spin-item-0-2-25 spin-fade-0-2-35"
      >
        <svg
          height="2.286px"
          viewBox="0 0 100 100"
          width="2.286px"
        >
          <circle
            cx="50"
            cy="50"
            r="50"
          />
        </svg>
      </div>
      <div
        class="spin-item-0-2-25 spin-fade-0-2-35"
      >
        <svg
          height="2.286px"
          viewBox="0 0 100 100"
          width="2.286px"
        >
          <circle
            cx="50"
            cy="50"
            r="50"
          />
        </svg>
      </div>
      <div
        class="spin-item-0-2-25 spin-fade-0-2-35"
      >
        <svg
          height="2.286px"
          viewBox="0 0 100 100"
          width="2.286px"
        >
          <circle
            cx="50"
            cy="50"
            r="50"
          />
        </svg>
      </div>
      <div
        class="spin-item-0-2-25 spin-fade-0-2-35"
      >
        <svg
          height="2.286px"
          viewBox="0 0 100 100"
          width="2.286px"
        >
          <circle
            cx="50"
            cy="50"
            r="50"
          />
        </svg>
      </div>
      <div
        class="spin-item-0-2-25 spin-fade-0-2-35"
      >
        <svg
          height="2.286px"
          viewBox="0 0 100 100"
          width="2.286px"
        >
          <circle
            cx="50"
            cy="50"
            r="50"
          />
        </svg>
      </div>
    </div>
    <div
      class="spin-tip-0-2-40"
    >
      <span
        style="font-size: 12px; font-weight: 300;"
      >
        fading-circle
      </span>
    </div>
  </div>
  <div
    class="spin-content-0-2-37 spin-vertical-0-2-41"
  >
    <div
      class="spin-fourDots-0-2-28 spin-spin-0-2-36"
      style="width: 16px; height: 16px;"
    >
      <div
        class="spin-item-0-2-25"
      >
        <svg
          viewBox="0 0 100 100"
        >
          <circle
            cx="50"
            cy="50"
            r="50"
          />
        </svg>
      </div>
      <div
        class="spin-item-0-2-25"
      >
        <svg
          viewBox="0 0 100 100"
        >
          <circle
            cx="50"
            cy="50"
            r="50"
          />
        </svg>
      </div>
      <div
        class="spin-item-0-2-25"
      >
        <svg
          viewBox="0 0 100 100"
        >
          <circle
            cx="50"
            cy="50"
            r="50"
          />
        </svg>
      </div>
      <div
        class="spin-item-0-2-25"
      >
        <svg
          viewBox="0 0 100 100"
        >
          <circle
            cx="50"
            cy="50"
            r="50"
          />
        </svg>
      </div>
    </div>
    <div
      class="spin-tip-0-2-40"
    >
      <span
        style="font-size: 12px; font-weight: 300;"
      >
        four-dots
      </span>
    </div>
  </div>
  <div
    class="spin-content-0-2-37 spin-vertical-0-2-41"
  >
    <div
      class="spin-plane-0-2-29"
      style="width: 16px; height: 16px;"
    />
    <div
      class="spin-tip-0-2-40"
    >
      <span
        style="font-size: 12px; font-weight: 300;"
      >
        plane
      </span>
    </div>
  </div>
  <div
    class="spin-content-0-2-37 spin-vertical-0-2-41"
  >
    <div
      class="spin-pulse-0-2-30"
      style="width: 16px; height: 16px;"
    />
    <div
      class="spin-tip-0-2-40"
    >
      <span
        style="font-size: 12px; font-weight: 300;"
      >
        pulse
      </span>
    </div>
  </div>
  <div
    class="spin-content-0-2-37 spin-vertical-0-2-41"
  >
    <div
      class="spin-ring-0-2-31"
      style="width: 16px; height: 16px; border-width: 1.6px; font-size: 1.6px;"
    />
    <div
      class="spin-tip-0-2-40"
    >
      <span
        style="font-size: 12px; font-weight: 300;"
      >
        ring
      </span>
    </div>
  </div>
  <div
    class="spin-content-0-2-37 spin-vertical-0-2-41"
  >
    <div
      class="spin-fadingCircle-0-2-26 spin-spin-0-2-36"
      style="width: 16px; height: 16px;"
    >
      <div
        class="spin-item-0-2-25 spin-scaleCircle-0-2-27"
      >
        <svg
          height="2.286px"
          viewBox="0 0 100 100"
          width="2.286px"
        >
          <circle
            cx="50"
            cy="50"
            r="50"
          />
        </svg>
      </div>
      <div
        class="spin-item-0-2-25 spin-scaleCircle-0-2-27"
      >
        <svg
          height="2.286px"
          viewBox="0 0 100 100"
          width="2.286px"
        >
          <circle
            cx="50"
            cy="50"
            r="50"
          />
        </svg>
      </div>
      <div
        class="spin-item-0-2-25 spin-scaleCircle-0-2-27"
      >
        <svg
          height="2.286px"
          viewBox="0 0 100 100"
          width="2.286px"
        >
          <circle
            cx="50"
            cy="50"
            r="50"
          />
        </svg>
      </div>
      <div
        class="spin-item-0-2-25 spin-scaleCircle-0-2-27"
      >
        <svg
          height="2.286px"
          viewBox="0 0 100 100"
          width="2.286px"
        >
          <circle
            cx="50"
            cy="50"
            r="50"
          />
        </svg>
      </div>
      <div
        class="spin-item-0-2-25 spin-scaleCircle-0-2-27"
      >
        <svg
          height="2.286px"
          viewBox="0 0 100 100"
          width="2.286px"
        >
          <circle
            cx="50"
            cy="50"
            r="50"
          />
        </svg>
      </div>
      <div
        class="spin-item-0-2-25 spin-scaleCircle-0-2-27"
      >
        <svg
          height="2.286px"
          viewBox="0 0 100 100"
          width="2.286px"
        >
          <circle
            cx="50"
            cy="50"
            r="50"
          />
        </svg>
      </div>
      <div
        class="spin-item-0-2-25 spin-scaleCircle-0-2-27"
      >
        <svg
          height="2.286px"
          viewBox="0 0 100 100"
          width="2.286px"
        >
          <circle
            cx="50"
            cy="50"
            r="50"
          />
        </svg>
      </div>
      <div
        class="spin-item-0-2-25 spin-scaleCircle-0-2-27"
      >
        <svg
          height="2.286px"
          viewBox="0 0 100 100"
          width="2.286px"
        >
          <circle
            cx="50"
            cy="50"
            r="50"
          />
        </svg>
      </div>
      <div
        class="spin-item-0-2-25 spin-scaleCircle-0-2-27"
      >
        <svg
          height="2.286px"
          viewBox="0 0 100 100"
          width="2.286px"
        >
          <circle
            cx="50"
            cy="50"
            r="50"
          />
        </svg>
      </div>
      <div
        class="spin-item-0-2-25 spin-scaleCircle-0-2-27"
      >
        <svg
          height="2.286px"
          viewBox="0 0 100 100"
          width="2.286px"
        >
          <circle
            cx="50"
            cy="50"
            r="50"
          />
        </svg>
      </div>
      <div
        class="spin-item-0-2-25 spin-scaleCircle-0-2-27"
      >
        <svg
          height="2.286px"
          viewBox="0 0 100 100"
          width="2.286px"
        >
          <circle
            cx="50"
            cy="50"
            r="50"
          />
        </svg>
      </div>
      <div
        class="spin-item-0-2-25 spin-scaleCircle-0-2-27"
      >
        <svg
          height="2.286px"
          viewBox="0 0 100 100"
          width="2.286px"
        >
          <circle
            cx="50"
            cy="50"
            r="50"
          />
        </svg>
      </div>
    </div>
    <div
      class="spin-tip-0-2-40"
    >
      <span
        style="font-size: 12px; font-weight: 300;"
      >
        scale-circle
      </span>
    </div>
  </div>
  <div
    class="spin-content-0-2-37 spin-vertical-0-2-41"
  >
    <div
      class="spin-threeBounce-0-2-32 spin-spin-0-2-36"
      style="width: 32px; height: auto;"
    >
      <div
        class="spin-item-0-2-25"
      >
        <svg
          height="8px"
          viewBox="0 0 100 100"
          width="8px"
        >
          <circle
            cx="50"
            cy="50"
            r="50"
          />
        </svg>
      </div>
      <div
        class="spin-item-0-2-25"
      >
        <svg
          height="8px"
          viewBox="0 0 100 100"
          width="8px"
        >
          <circle
            cx="50"
            cy="50"
            r="50"
          />
        </svg>
      </div>
      <div
        class="spin-item-0-2-25"
      >
        <svg
          height="8px"
          viewBox="0 0 100 100"
          width="8px"
        >
          <circle
            cx="50"
            cy="50"
            r="50"
          />
        </svg>
      </div>
    </div>
    <div
      class="spin-tip-0-2-40"
    >
      <span
        style="font-size: 12px; font-weight: 300;"
      >
        three-bounce
      </span>
    </div>
  </div>
  <div
    class="spin-content-0-2-37 spin-vertical-0-2-41"
  >
    <div
      class="spin-wave-0-2-33 spin-spin-0-2-36"
      style="width: 16px; height: 16px;"
    >
      <div
        class="spin-item-0-2-25"
        style="width: 2px; margin-right: 1px;"
      />
      <div
        class="spin-item-0-2-25"
        style="width: 2px; margin-right: 1px;"
      />
      <div
        class="spin-item-0-2-25"
        style="width: 2px; margin-right: 1px;"
      />
      <div
        class="spin-item-0-2-25"
        style="width: 2px; margin-right: 1px;"
      />
      <div
        class="spin-item-0-2-25"
        style="width: 2px; margin-right: 1px;"
      />
    </div>
    <div
      class="spin-tip-0-2-40"
    >
      <span
        style="font-size: 12px; font-weight: 300;"
      >
        wave
      </span>
    </div>
  </div>
  <div
    class="spin-content-0-2-37 spin-vertical-0-2-41"
  >
    <div
      class="spin-chasingRing-0-2-34 spin-spin-0-2-36"
      style="width: 16px; height: 16px;"
    >
      <div
        class="spin-item-0-2-25"
        style="background-color: transparent; border-width: 1.6px;"
      />
      <div
        class="spin-item-0-2-25"
        style="background-color: transparent; border-width: 1.6px;"
      />
      <div
        class="spin-item-0-2-25"
        style="background-color: transparent; border-width: 1.6px;"
      />
      <div
        class="spin-item-0-2-25"
        style="background-color: transparent; border-width: 1.6px;"
      />
    </div>
    <div
      class="spin-tip-0-2-40"
    >
      <span
        style="font-size: 12px; font-weight: 300;"
      >
        chasing-ring
      </span>
    </div>
  </div>
</div>
`;

exports[`Spin[Mode] should render correctly  1`] = `
<div
  style="display: flex; align-items: center; gap: 24px;"
>
  <div
    style="width: 100px;"
  >
    <div
      class="spin-content-0-2-37 spin-vertical-0-2-41"
    >
      <div
        class="spin-ring-0-2-31"
        style="width: 16px; height: 16px; border-width: 1.6px; font-size: 1.6px;"
      />
      <div
        class="spin-tip-0-2-40"
      >
        <span
          style="font-size: 12px; font-weight: 300;"
        >
          Loading...
        </span>
      </div>
    </div>
  </div>
  <div
    style="width: 100px;"
  >
    <div
      class="spin-content-0-2-37 spin-horizontal-0-2-42"
    >
      <div
        class="spin-ring-0-2-31"
        style="width: 16px; height: 16px; border-width: 1.6px; font-size: 1.6px;"
      />
      <div
        class="spin-tip-0-2-40"
      >
        <span
          style="font-size: 12px; font-weight: 300;"
        >
          Loading...
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`Spin[Tip] should render correctly  1`] = `
<div
  style="width: 200px;"
>
  <div
    class="spin-content-0-2-37 spin-vertical-0-2-41"
  >
    <div
      class="spin-ring-0-2-31"
      style="width: 24px; height: 24px; border-width: 2.4px; font-size: 2.4px;"
    />
    <div
      class="spin-tip-0-2-40"
    >
      <div
        style="font-weight: 300; font-size: 14px;"
      >
        This may take a while...
      </div>
    </div>
  </div>
</div>
`;
